### PR TITLE
Enum Model-Trait

### PR DESCRIPTION
**Pull request by NeroAzure**
### Fixed
- the `lint:phpmd`-script no longer searches for a non existing directory
- localization for the `EnumerationValue`-message

### Changed
- `__toString` now returns the value and not the key of the member

### Added
- `defaultMember` method to `Enumeration`
- `HasEnums` trait for laravel models
- a bunch of additional tests

### Updated
- added new information to readme
- changelog

### CODE REVIEW

File: tests/feature/ModelHasEnumsTraitTest.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->model = new TestModel;
+    }
+
+    /**
+     * Check if the default member gets returned from attribute
+     * access when the attribute is null (or a value not represented by the enum)
```
Comment (krns) [inline tests/feature/ModelHasEnumsTraitTest.php]:
```suggestion
     * access when the attribute is null (or a value not represented by the enum).
```

File: tests/feature/ModelHasEnumsTraitTest.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            $this->model->test = 'test';
+            $this->assertTrue(false, 'undefined value was set.');
+        } catch (UndefinedMemberException $e) {
+            $this->assertTrue(true);
+        }
+    }
+
+    /**
+     * Check if the validation when setting an enum-property
+     * with a correct skalar value throws no exception.
```
Comment (krns) [inline tests/feature/ModelHasEnumsTraitTest.php]:
```suggestion
     * with a correct scalar value throws no exception.
```

File: tests/unit/HasEnumsTraitMethodsTest.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    public function testGetAttribute(): void
+    {
+        $this->assertEquals(TestEnum::defaultMember(), $this->model->getAttribute('test'));
+
+        $this->model->test = TestEnum::TEST3();
+
+        $this->assertEquals(TestEnum::TEST3(), $this->model->getAttribute('test'));
+
+        $this->assertNull($this->model->getAttribute('test2'));
+        $this->assertNull($this->model->getAttribute('test3'));
```
Comment (krns) [inline tests/unit/HasEnumsTraitMethodsTest.php]: Maybe choose a better name for the parameter (takes effect for the PR)

Comment (krns) [reply tests/unit/HasEnumsTraitMethodsTest.php]: Think about putting some `test` occurrences  aside and use semantically better values.

Comment (NeroAzure) [reply tests/unit/HasEnumsTraitMethodsTest.php]: Changed all occurrences of TestEnum to UserRole and TestEnum2 to FruitType, with corresponding values for those enums.

**Pull request merged**